### PR TITLE
UEFITool NE 59

### PR DESCRIPTION
**Pull request by vampirecat35**
- Updates for QT 5 and QT 6
- Deprecation fixes
- Code fixes

### CODE REVIEW

**Comment by vit9696** [comment]
Hello, please update the compatibility with the CI. We still require Qt 5.6 as a minimum for LGPL 2.0 license reasons.

**Comment by vampirecat35** [comment]
> Hello, please update the compatibility with the CI. We still require Qt 5.6 as a minimum for LGPL 2.0 license reasons.

I used Qt-5.15.2 and Qt-6.0.3
I tested it on macOS.

**Comment by vampirecat35** [comment]
> Hello, please update the compatibility with the CI. We still require Qt 5.6 as a minimum for LGPL 2.0 license reasons.

Done.
I added a QT version check.
If 5.11 or older use .width
Otherwise use new method.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: CI
+x@name: CI
```
Comment (vit9696) [inline .github/workflows/main.yml:1]: You broke CI completely here by accident.

File: UEFITool/uefitool.pro [hunk 2]
```diff
@@ -8,6 +8,8 @@ DEFINES += "U_ENABLE_NVRAM_PARSING_SUPPORT"
 DEFINES += "U_ENABLE_ME_PARSING_SUPPORT"
 DEFINES += "U_ENABLE_GUID_DATABASE_SUPPORT"
 
+greaterThan(QT_MAJOR_VERSION, 5): DEFINES += "QT_VERSION_MAJOR=6"
```
Comment (vit9696) [inline UEFITool/uefitool.pro:11]: Are you sure there is no builtin macro to check against Qt version?

File: .github/workflows/main.yml (hunk 1 above)
Comment (vampirecat35) [reply .github/workflows/main.yml]: Fixed.
Accidentally edited the file.

**Comment by vampirecat35** [comment]
CI is fixed and build success.
Also tested on newer QT versions and success too.

File: UEFITool/uefitool.pro (hunk 2 above)
Comment (vampirecat35) [reply UEFITool/uefitool.pro]: True.
This isn't needed.
Pushed commit

File: UEFITool/ffsfinder.cpp [hunk 3]
```diff
@@ -29,7 +29,12 @@ USTATUS FfsFinder::findHexPattern(const UModelIndex & index, const UByteArray &
 
     bool hasChildren = (model->rowCount(index) > 0);
     for (int i = 0; i < model->rowCount(index); i++) {
+#ifdef _USE_DEPRECATED
```
Comment (vit9696) [inline UEFITool/ffsfinder.cpp:32]: Why cannot we use non-deprecated interfaces exclusively? Does Qt 5.6 not support them? Who does define `_USE_DEPRECATED`?

Comment (vampirecat35) [reply UEFITool/ffsfinder.cpp:32]: I'll do a version check.
Fixing in a minute

**Comment by vampirecat35** [comment]
Fixed older QT versions.
Newer ones will use the new code.

File: common/treemodel.h [hunk 4]
```diff
@@ -86,7 +88,7 @@ class UModelIndex
 #if defined(QT_CORE_LIB)
 class TreeModel : public QAbstractItemModel
 {
-    Q_OBJECT
+    //Q_OBJECT
```
Comment (vit9696) [inline common/treemodel.h:91]: Better just remove than comment out.

File: common/treemodel.h [hunk 5]
```diff
@@ -35,7 +35,9 @@ enum ItemFixedState {
 #include "types.h"
 #include "treeitem.h"
 
+#ifndef UModelIndex
```
Comment (vit9696) [inline common/treemodel.h:38]: Why is that one needed?

File: common/ustring.cpp [hunk 6]
```diff
@@ -19,7 +19,7 @@ UString usprintf(const char* fmt, ...)
     UString msg;
     va_list vl;
     va_start(vl, fmt);
-    msg.vsprintf(fmt, vl);
+    msg = msg.vasprintf(fmt, vl);
```
Comment (vit9696) [inline common/ustring.cpp:22]: Why is this necessary?

**Comment by vit9696** [comment]
Looks mostly good, spotted a few more weird points, but should be ready to merge soon after.

**Comment by vampirecat35** [comment]
I added it using a version check.
This way older than QT 5.6 also builds.

File: common/ustring.cpp (hunk 6 above)
Comment (vampirecat35) [reply common/ustring.cpp:22]: Newer QT versions like QT 6 don't implement .vsprintf.
This thus needs the newer method.

File: common/treemodel.h (hunk 5 above)
Comment (vampirecat35) [reply common/treemodel.h:38]: Removing...

File: common/treemodel.h (hunk 4 above)
Comment (vampirecat35) [reply common/treemodel.h]: Removed

**Comment by vampirecat35** [comment]
Image of UEFITool on macOS Big Sur using QT 6
[https://i.ibb.co/WnFR9YZ/UEFITool-NE-59-mac-OS-QT6.png](UEFITool QT6)

**Comment by vampirecat35** [comment]
Removed Xcode projects, copied new icon file, added uefitool.entitlements for use with code sign
codesign -fs "Apple Development" --entitlements uefitool.entitlements was used in my case.
This is needed after macdeployqt for adding sandbox and signing.

**Comment by vampirecat35** [comment]
Done with all QT 6 updates.
It now builds without a warning for QT 6.
